articles/code-management/about-code-management-csharp/about-code-management-csharp.md

```diff
@@ -0,0 +1,98 @@
+* `Ignore` - It has **no** control over the file and will not generate or overwrite anything in it.
+
+> [!NOTE]
+> Comments that are made by the developer (not the Code Automation) outside the scope of a `namespace` will not be modified in any way even if `Mode` is set to `Fully`.
+
+## Override different parts using `[IntentManaged]`
+
+Within a C# source file, one may choose to override certain behaviour of how Intent Architect performs code automation by inserting a `[IntentManaged]` attribute right above a class, property, field or method (or various other language elements).
+
+For example, adding this attribute above a method will look something like this:
+
+```csharp
+[IntentManaged(Mode.Ignore)]
+public void ChangeCountry(string country)
+{
+    throw new NotImplementedException();
+}
+```
+
+It will instruct Intent Architect not to modify (or remove) this method in any way during the Software Factory Execution.
+
+This attribute does have additional properties of configuration which gives a developer finer grain of control. Below can be seen the attribute written in C# in its entirety, followed by more in depth descriptions of what each piece does (for this example within the context of a method).
+
+```csharp
+[IntentManaged(Mode.Merge, Body = Mode.Ignore, Signature = Mode.Fully, Comments = Mode.Fully, Attributes = Mode.Fully)]
+```
+> [!NOTE]
+> The `Mode` configuration works exactly the same as with the `DefaultIntentManaged` attribute, except that the scope is now on a method level and not a file level.
+
+| Component       | Mandatory | Description                                                                                                                                                         |
+|-----------------|-----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| First parameter | Yes       | Base setting which targets the entire method (including the method name, parameter list, attributes, comments and method body).                                     |
+| Body            | No        | Overrides the base setting in order for the method body to be overwritten by Intent Architect or not.                                                               |
+| Signature       | No        | Overrides the base setting in order for the method name, parameter list, attributes and comments to be overwritten by Intent Architect or not.                      |
+| Comments        | No        | Overrides the base setting **and** signature setting in order for the comments located right above the method to be overwritten by Intent Architect or not.         |
+| Attributes      | No        | Overrides the base setting **and** signature setting in order for the attributes (other than the `IntentManaged` one) to be overwritten by Intent Architect or not. |
```

Comment:
There is a lot of great information in here, but it shouldn't be specifically in the context of a method, it should be general to use of the `[IntentManaged]` attribute.

We need an additional matrix to describe what "Body" and "Signature" refers to for different syntax node types, for example, for example, for a class the body refers to its methods, and for a field it refers to its value, for a method, it refers to its body, etc. Then do something similar for "signature".

I don't think we need to worry about comment and attributes in the matrix as they should be self explanatory.